src/components/item/header/ItemHeaderActions.tsx

```diff
@@ -69,33 +127,128 @@ const ItemHeaderActions = ({ item }: Props): JSX.Element => {
+                canEdit={showEditButton}
+              >
+                {shareActions}
+              </MobileItemMenu>
+            ) : (
+              <ItemMenu item={item} canMove={false} canEdit={showEditButton} />
+            )}
+          </Box>
 
-          <ShareButton itemId={item.id} />
-          <ChatboxButton
-            tooltip={translateBuilder(BUILDER.ITEM_CHATBOX_TITLE)}
-            id={ITEM_CHATBOX_BUTTON_ID}
-            onClick={onClickChatbox}
-          />
-          {canAdmin && <PublishButton itemId={item.id} />}
+          {!isMobile && shareActions}
+          {isMobile && <Divider />}
+          {!isMobile && (
+            <ChatboxButton
+              tooltip={translateBuilder(BUILDER.ITEM_CHATBOX_TITLE)}
+              id={ITEM_CHATBOX_BUTTON_ID}
+              onClick={onClickChatbox}
+            />
+          )}
         </>
       );
 
       return (
         <>
           {openedActionTabId !== ItemActionTabs.Settings && activeActions}
-          {canEdit && <ItemSettingsButton id={item.id} />}
-          <DownloadButton id={item.id} name={item.name} />
+          <IconButtonWithText
+            isMobile={isMobile}
+            text={translateBuilder(BUILDER.DOWNLOAD_ITEM_BUTTON)}
+            onClick={closeDrawer}
+          >
+            <DownloadButton id={item.id} name={item.name} />
+          </IconButtonWithText>
+          {canEdit && (
+            <IconButtonWithText
+              isMobile={isMobile}
+              text={translateBuilder(BUILDER.SETTINGS_TITLE)}
+              onClick={closeDrawer}
+            >
+              <ItemSettingsButton id={item.id} />
+            </IconButtonWithText>
+          )}
         </>
       );
     }
     return null;
   };
 
+  if (isMobile) {
+    return item ? (
+      <>
+        <Box display="flex" sx={{ '& button': { padding: 0.5 } }}>
+          <ItemMetadataButton />
+          <ModeButton />
+          <ChatboxButton
+            tooltip={translateBuilder(BUILDER.ITEM_CHATBOX_TITLE)}
+            id={ITEM_CHATBOX_BUTTON_ID}
+            onClick={onClickChatbox}
+          />
+          <IconButton onClick={toggleActionsDrawer}>
+            <MoreVertIcon />
+          </IconButton>
+        </Box>
+
+        <SwipeableDrawer
+          anchor="bottom"
+          open={isItemActionsDrawerOpen}
+          onClose={closeDrawer}
+          onOpen={() => setIsItemActionsDrawerOpen(true)}
+          PaperProps={{ sx: { maxHeight: '80vh', paddingY: 2 } }}
+        >
+          <Typography px={3} pb={1} variant="h5">
+            {item.name}
+          </Typography>
+          <Divider sx={{ borderColor: theme.palette.grey[400] }} />
+          <Box
+            display="flex"
+            flexDirection="column"
+            sx={{ maxHeight: '100%', overflow: 'auto', marginTop: 1 }}
+          >
+            {renderItemActions()}
+          </Box>
+        </SwipeableDrawer>
+      </>
+    ) : (
+      <ModeButton />
+    );
+  }
```

Comment:
I think this component is too complicated. You use `isMobile` everywhere in all different ways. It would be simpler to have a new component `MobileItemHeaderActions` that you display on `isMobile` (defined in `ItemHeader.tsx`) and keep the old `ItemHeaderActions` file.